airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -576,17 +590,22 @@ def stream_slices(
         )
         if incremental_slices:
             parent_records = HttpSubStream.stream_slices(self, sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state)
-            yield from (slice | rec for rec in parent_records for slice in incremental_slices)
+            yield from (
+                slice_ | rec
+                for rec in parent_records
+                for slice_ in incremental_slices
+                if isinstance(slice_, dict) and isinstance(rec, dict)
```

Comment:
I find this a bit dangerous as we could want to change this to any class that support the interface `__get__` and this would start failing. Is there are reason why we need to validate that it is an instance of a dict?